src/app/screens/sendBrc20/index.tsx

```diff
@@ -1,91 +1,66 @@
-import styled from 'styled-components';
-import BigNumber from 'bignumber.js';
-import { useEffect, useState } from 'react';
-import { useTranslation } from 'react-i18next';
-import { useLocation, useNavigate } from 'react-router-dom';
+import BottomBar from '@components/tabBar';
+import TopRow from '@components/topRow';
+import useBtcFeeRate from '@hooks/useBtcFeeRate';
+import useWalletSelector from '@hooks/useWalletSelector';
 import {
-  createBrc20TransferOrder,
-  getBtcFiatEquivalent,
-  signBtcTransaction,
-  ErrorCodes,
+  Brc20TransferEstimateFeesParams,
+  ConfirmBrc20TransferState,
+  SendBrc20TransferState,
+} from '@utils/brc20';
+import { BRC20ErrorCode } from '@secretkeylabs/xverse-core/transactions/brc20';
+import {
+  UTXO,
+  brc20TransferEstimateFees,
+  getNonOrdinalUtxo,
+  validateBtcAddress,
+  CoreError,
 } from '@secretkeylabs/xverse-core';
-import { Recipient } from '@secretkeylabs/xverse-core/transactions/btc';
+import { getFtTicker } from '@utils/tokens';
+import { replaceCommaByDot } from '@utils/helper';
+import { useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
 import { useResetUserFlow } from '@hooks/useResetUserFlow';
-import useWalletSelector from '@hooks/useWalletSelector';
-import TopRow from '@components/topRow';
-import BottomBar from '@components/tabBar';
-import ActionButton from '@components/button';
+import { useTranslation } from 'react-i18next';
 import Brc20TransferForm from './brc20TransferForm';
-import Brc20TransferInfo from './brc20TransferInfo';
-
-const BRC20TokenTagContainer = styled.div((props) => ({
-  display: 'flex',
-  flexDirection: 'row',
-  justifyContent: 'center',
-  marginTop: props.theme.spacing(3),
-}));
-
-const BRC20TokenTag = styled.div((props) => ({
-  background: props.theme.colors.white[400],
-  borderRadius: 40,
-  width: 54,
-  height: 19,
-  padding: '2px 6px',
-  h1: {
-    ...props.theme.body_bold_l,
-    fontSize: 11,
-    color: props.theme.colors.background.elevation0,
-  },
-}));
-
-interface ButtonProps {
-  enabled: boolean;
-}
-
-const SendButtonContainer = styled.div<ButtonProps>((props) => ({
-  paddingBottom: props.theme.spacing(12),
-  paddingTop: props.theme.spacing(4),
-  marginLeft: '5%',
-  marginRight: '5%',
-  opacity: props.enabled ? 1 : 0.6,
-}));
 
 function SendBrc20Screen() {
-  const { t } = useTranslation('translation');
+  const { t } = useTranslation('translation', { keyPrefix: 'SEND_BRC20' });
   const navigate = useNavigate();
-  const {
-    btcAddress,
-    ordinalsAddress,
-    selectedAccount,
-    seedPhrase,
-    network,
-    btcFiatRate,
-    brcCoinsList,
-  } = useWalletSelector();
+  const location = useLocation();
+  const { btcAddress, ordinalsAddress, network, brcCoinsList } = useWalletSelector();
+  const { data: feeRate } = useBtcFeeRate();
   const [amountError, setAmountError] = useState('');
   const [amountToSend, setAmountToSend] = useState('');
```

Comment:
This would probably be easier to keep as a number instead of casting it everywhere 😋 

```suggestion
  const [amountToSend, setAmountToSend] = useState(0);
```

Reply:
It would have the opposite problem if I store it as number, where I'd have to cast it to String to set the input value.

I usually like to keep the state for input boxes as string, to match the html input value field.
pros:
- default unfilled input boxes can have value="", whereas value="0" is often not a good default
- if you cast default 0 back to string, you get a truthy value, which makes it more verbose trying to validate whether the input was filled or not: 
![image](https://github.com/secretkeylabs/xverse-web-extension/assets/6109710/37e52b00-ddc2-4015-a666-32909a3f9088)
- casting to number is opt-in, and occurs just in time (not prematurely) for, e.g. validation rules, or passing on to function params.

Reply:
Ah, good point 👍 I never thought about the cast back to string before.